docs/reference-implementation/Managed-prometheus.md

```diff
@@ -0,0 +1,43 @@
+
+As you can see, Prometheus, Prometheus metrics and especially the Prometheus Query Language (PromQL) are a defacto standard in the cloud native space. What options do we have to leverage these metrics? The existing Azure Monitor agent already supports scraping prometheus metrics and store them in Log Analytics. This was already quiet helpful, but, Log Analytics is a quiet expensive storage and it supports Kusto Query Language (KQL) but not the Prometheus Query Language (PromQL). And this can become a problem, especially when using Prometheus across environments and platforms.
+
+## Managed Prometheus
+
+To use Prometheus you could already have deployed a Prometheus server on your own to scrape the metrics endpoints and provide the data to visualization tools like Grafana. But Prometheus is a beast by itself, you've to take care of the infrastructure, operations, maintenance and so on. This was now made easier with the availability of the **Azure Monitor managed service for Prometheus**. This managed service offering is compliant with the open-source project prometheus, but it abstracts the underlying infrastructure away. It consists of the following components:
+
+- The 'ama-agent' is deployed on Kubernetes to gather the metrics.
+- An **Azure Monitor workspace** where the collected data is stored.
+- **Data collection rules** to determine what data to collect.
+- **Data collection endpoints** define the target to sent the data to for analysis and storage.
+- **Data collection rule associations** to link rules and resources they apply to.
+
+This once set up provides you with a query endpoint that can be leveraged by for example Grafana to feed dashboards.
+
+## How to fit Prometheus into Azure Mission-Critical?
+
+You might wonder how these additional resources are getting deployed and where to put them. The answer is easy, the mission-critical architecture pattern separates between global and regional components as well as between stateful and stateless components.
+
+The Azure Monitor workspace that stores the collected data is getting deployed into the `<prefix>-monitoring-rg`. One instance per region used.
+
+The other resources, like the Data collection rule, endpoint and rule association is deployed alongside the regional stamps. They're ephemeral and share the same lifecycle as the stamp. When a new stamp is getting deployed, new rules, endpoints and associations are deployed and the old ones are getting destroyed.
+
```

Comment:
Should we add something about data retention? I think data retention on AzMon WS is 18 month... can we easily off load metrics to a cold storage?

Reply:
Please make a suggestion. I don't want to overcomplicate things here. Maybe a link to https://learn.microsoft.com/en-us/azure/azure-monitor/essentials/azure-monitor-workspace-overview#data-considerations?